### ISSUE TITLE
Invalid configuration for `@typescript-eslint/member-ordering`

### ISSUE THREAD

**Issue by jamesacarr** [xojs/eslint-config-xo-typescript#67]
Using the latest version of `@typescript-eslint` (5.45.0), and error is being raised in the configuration of `@typescript-eslint/member-ordering`.

```
Oops! Something went wrong! :(

ESLint: 8.28.0

Error: .eslintrc.js#overrides[1] » eslint-config-xo-typescript/space » /<censored>/node_modules/eslint-config-xo-typescript/index.js:
	Configuration for rule "@typescript-eslint/member-ordering" is invalid:
	Value ["signature","public-static-field","public-static-method","protected-static-field","protected-static-method","private-static-field","private-static-method","static-field","static-method","public-decorated-field","public-instance-field","public-abstract-field","public-field","protected-decorated-field","protected-instance-field","protected-abstract-field","protected-field","private-decorated-field","private-instance-field","private-abstract-field","private-field","instance-field","abstract-field","decorated-field","field","public-constructor","protected-constructor","private-constructor","constructor","public-decorated-method","public-instance-method","public-abstract-method","public-method","protected-decorated-method","protected-instance-method","protected-abstract-method","protected-method","private-decorated-method","private-instance-method","private-abstract-method","private-method","instance-method","abstract-method","decorated-method","method"] should be string.
	Value ["signature","public-static-field","public-static-method","protected-static-field","protected-static-method","private-static-field","private-static-method","static-field","static-method","public-decorated-field","public-instance-field","public-abstract-field","public-field","protected-decorated-field","protected-instance-field","protected-abstract-field","protected-field","private-decorated-field","private-instance-field","private-abstract-field","private-field","instance-field","abstract-field","decorated-field","field","public-constructor","protected-constructor","private-constructor","constructor","public-decorated-method","public-instance-method","public-abstract-method","public-method","protected-decorated-method","protected-instance-method","protected-abstract-method","protected-method","private-decorated-method","private-instance-method","private-abstract-method","private-method","instance-method","abstract-method","decorated-method","method"] should be equal to one of the allowed values.
	Value "private-abstract-field" should be equal to one of the allowed values.
	Value "private-abstract-field" should be array.
	Value "private-abstract-field" should match exactly one schema in oneOf.
	Value ["signature","public-static-field","public-static-method","protected-static-field","protected-static-method","private-static-field","private-static-method","static-field","static-method","public-decorated-field","public-instance-field","public-abstract-field","public-field","protected-decorated-field","protected-instance-field","protected-abstract-field","protected-field","private-decorated-field","private-instance-field","private-abstract-field","private-field","instance-field","abstract-field","decorated-field","field","public-constructor","protected-constructor","private-constructor","constructor","public-decorated-method","public-instance-method","public-abstract-method","public-method","protected-decorated-method","protected-instance-method","protected-abstract-method","protected-method","private-decorated-method","private-instance-method","private-abstract-method","private-method","instance-method","abstract-method","decorated-method","method"] should be object.
	Value ["signature","public-static-field","public-static-method","protected-static-field","protected-static-method","private-static-field","private-static-method","static-field","static-method","public-decorated-field","public-instance-field","public-abstract-field","public-field","protected-decorated-field","protected-instance-field","protected-abstract-field","protected-field","private-decorated-field","private-instance-field","private-abstract-field","private-field","instance-field","abstract-field","decorated-field","field","public-constructor","protected-constructor","private-constructor","constructor","public-decorated-method","public-instance-method","public-abstract-method","public-method","protected-decorated-method","protected-instance-method","protected-abstract-method","protected-method","private-decorated-method","private-instance-method","private-abstract-method","private-method","instance-method","abstract-method","decorated-method","method"] should match exactly one schema in oneOf.
```

The configuration (below) includes two values that are invalid - `private-abstract-field` and `private-abstract-method`. Removing these two entries from the array fixes the issue.

https://github.com/xojs/eslint-config-xo-typescript/blob/6d5aba90d656678efd5150b41e663b8931a2c592/index.js#L294-L358

Removing these two entries from the array fixes the issue.
